components/inputs/input-number.js

```diff
@@ -99,42 +92,84 @@ class InputNumber extends SkeletonMixin(FormElementMixin(LocalizeCoreElement(Lit
 		this.maxExclusive = false;
 		this.minExclusive = false;
 		this.required = false;
-		this.trailingZeroes = false;
 
 		this._formattedValue = '';
 		this._inputId = getUniqueId();
-		this._numDecimalDigits = 0;
+		this._trailingZeroes = false;
+		this._valueTrailingZeroes = '';
+	}
+
+	get maxFractionDigits() {
+		return this._maxFractionDigits ?? Math.max(this.minFractionDigits, 3);
```

Comment:
I updated these to always return a value, instead of previously where they'd default to `undefined`. I felt like this better reflected what was going on, since the Intl library would still be rounding to 3 decimals by default.